File: backend/app/api/sessions.py
"""sessions.py - 会话管理 API"""
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import List
from ..core.session_store import (
    list_sessions,
    get_session,
    save_session,
    delete_session,
    rename_session,
    get_session_messages,
    save_session_messages,
)
import time

router = APIRouter(prefix="/sessions", tags=["sessions"])


class CreateSessionReq(BaseModel):
    title: str = "新对话"
    id: str = None


class RenameReq(BaseModel):
    title: str


class AddMessageReq(BaseModel):
    role: str
    content: str = ""
    msg: str = ""
    files: list = None


@router.get("")
def list_all():
    """列出所有会话"""
    return list_sessions()


@router.post("")
def create(req: CreateSessionReq):
    """新建会话"""
    sid = req.id or f"s{int(time.time() * 1000)}"
    data = save_session(sid, {"title": req.title, "messages": []})
    return data


@router.get("/{session_id}")
def read(session_id: str):
    """读取会话"""
    data = get_session(session_id)
    if not data:
        raise HTTPException(404, f"会话不存在: {session_id}")
    return data


@router.delete("/{session_id}")
def remove(session_id: str):
    """删除会话"""
    if not delete_session(session_id):
        raise HTTPException(404, f"会话不存在: {session_id}")
    return {"status": "ok", "deleted": session_id}


@router.patch("/{session_id}/title")
def update_title(session_id: str, req: RenameReq):
    """修改标题"""
    data = rename_session(session_id, req.title)
    if not data:
        raise HTTPException(404, f"会话不存在: {session_id}")
    return data


@router.get("/{session_id}/messages")
def list_messages(session_id: str):
    """获取消息列表"""
    return get_session_messages(session_id)


@router.post("/{session_id}/messages")
def add_message(session_id: str, req: AddMessageReq):
    """追加一条消息"""
    msgs = get_session_messages(session_id)
    msgs.append({
        "role": req.role,
        "content": req.content or req.msg,
        "files": req.files,
    })
    save_session_messages(session_id, msgs)
    return {"status": "ok", "count": len(msgs)}


@router.put("/{session_id}/messages")
def replace_messages(session_id: str, messages: List[dict]):
    """替换消息列表"""
    save_session_messages(session_id, messages)
    return {"status": "ok", "count": len(messages)}

File: frontend/src/services/sessions.ts
// sessions API - 历史对话管理
function encodeName(name: string): string {
  return encodeURIComponent(name)
}

export interface SessionInfo {
  id: string
  title: string
  created: number
  updated: number
  message_count?: number
}

export interface SessionMessage {
  role: string
  content?: string
  msg?: string
  files?: string[]
}

const BASE = ''

export async function listSessions(): Promise<SessionInfo[]> {
  const res = await fetch(`${BASE}/api/sessions`)
  if (!res.ok) throw new Error(`listSessions failed: ${res.status}`)
  return res.json()
}

export async function createSession(title = '新对话'): Promise<SessionInfo> {
  const res = await fetch(`${BASE}/api/sessions`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ title }),
  })
  if (!res.ok) throw new Error(`createSession failed: ${res.status}`)
  return res.json()
}

export async function getSession(id: string): Promise<any> {
  const res = await fetch(`${BASE}/api/sessions/${encodeName(id)}`)
  if (!res.ok) throw new Error(`getSession failed: ${res.status}`)
  return res.json()
}

export async function deleteSession(id: string): Promise<void> {
  const res = await fetch(`${BASE}/api/sessions/${encodeName(id)}`, {
    method: 'DELETE',
  })
  if (!res.ok) throw new Error(`deleteSession failed: ${res.status}`)
}

export async function renameSession(id: string, title: string): Promise<SessionInfo> {
  const res = await fetch(`${BASE}/api/sessions/${encodeName(id)}/title`, {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ title }),
  })
  if (!res.ok) throw new Error(`renameSession failed: ${res.status}`)
  return res.json()
}

export async function getSessionMessages(id: string): Promise<SessionMessage[]> {
  const res = await fetch(`${BASE}/api/sessions/${encodeName(id)}/messages`)
  if (!res.ok) throw new Error(`getSessionMessages failed: ${res.status}`)
  return res.json()
}

export async function saveSessionMessages(id: string, messages: SessionMessage[]): Promise<void> {
  const res = await fetch(`${BASE}/api/sessions/${encodeName(id)}/messages`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(messages),
  })
  if (!res.ok) throw new Error(`saveSessionMessages failed: ${res.status}`)
}